source/geometries/NextDemoSiPMBoard.cc

```diff
@@ -207,25 +209,59 @@ void NextDemoSiPMBoard::Construct()
 
 
   /// Mask Holes
+  G4String coating_name = "BOARD_COATING";
+  G4Material* tpb = MaterialsList::TPB();
+  tpb->SetMaterialPropertiesTable(OpticalMaterialProperties::TPB());
+  G4OpticalSurface* coating_opsurf =
+    new G4OpticalSurface(coating_name + "_OPSURF", glisur, ground,
+                         dielectric_dielectric, .01);
+
   G4VSolid* hole_solid = nullptr;
   G4String hole_name = "BOARD_MASK_HOLE";
   if (hole_type_ == "rounded"){
     hole_solid = new G4Tubs(hole_name, 0., hole_diam_/2., mask_thickn_/2., 0, 360.*deg);}
   else if (hole_type_ == "rectangular"){
     hole_solid = new G4Box(hole_name, hole_x_/2., hole_y_/2., mask_thickn_/2.);}
 
-  G4LogicalVolume* hole_logic =
-    new G4LogicalVolume(hole_solid, mother_gas, hole_name);
+  G4LogicalVolume* hole_logic = new G4LogicalVolume(hole_solid, mother_gas, hole_name);
 
   // Generate SiPM positions
   GenerateSiPMPositions();
 
   G4double sipm_posz = - mask_thickn_/2. + sipm_z_dim/2.;
 
-  new G4PVPlacement(sipm_rot, G4ThreeVector(0., 0., sipm_posz), sipm_->GetLogicalVolume(),
-                    sipm_->GetLogicalVolume()->GetName(), hole_logic,
-                    false, 0, false);
+  G4VPhysicalVolume* hole_phys_ = new G4PVPlacement(sipm_rot, G4ThreeVector(0., 0., sipm_posz), sipm_->GetLogicalVolume(),
+                                                    sipm_->GetLogicalVolume()->GetName(), hole_logic,
+                                                    false, 0, false);
+
+  if (hole_coated && (hole_type_ == "rectangular")){
+
+    G4Box* hole_coating_top = new G4Box("HOLE_COATING", hole_x_/2., coating_thickn_/2., mask_thickn_/2.);
+    G4Box* hole_coating_lat = new G4Box("HOLE_COATING", coating_thickn_/2., hole_y_/2., mask_thickn_/2.);
+
+    G4UnionSolid* hole_coating = new G4UnionSolid("HOLE_COATING", hole_coating_top, hole_coating_lat, 0,
```

Comment:
I agree that your solution would also be valid and probably easier to implement, and also the proper way to implement the coating for rounded holes. Nevertheless you can check that the sides are properly placed. In the figure below I show the coating placed inside the hole for a coating thickness of 1 mm (to improve visualization).
<img width="369" alt="coated sipm hole" src="https://user-images.githubusercontent.com/35993868/123650717-21f79880-d82b-11eb-98c6-1d7e49e48fd6.png">

Reply:
Cool. I still think it would be more efficient and simple with the `G4SubtractionSolid`, it would be just two `G4Box` one inside the other. I'm not sure if you would have any issue in the joint areas.

Reply:
What issue? It is a `G4UnionSolid` thus the overlapping is managed. There is no geometry overlap either.

Reply:
To move here the discussion in Slack: the best way to do it is putting a `G4Box` inside another one, no subtractions neither unions involved, which may slow down the tracking. The bigger box will be made of TPB and the smaller one of the gas of the mother volume. Also, if we ever need to do the same with the rounded holes, it can be done the same way.

Reply:
Since the SiPM resides in the hole, the two `G4Box` implementation you suggest would requiere to modify the volume were they are placed, and consequently the previous run configurations. The volumes are placed as intended and inner coating is only added if required, which  would not be the case with two boxes since one of them is made of tpb.

For external reader: we agreed via Slack discussion to maintain this implementation since is the minimal change possible. We would consider the change in the future if needed, for example, to implement rounded hole coatings.